crates/task-impls/src/quorum_proposal_recv/handlers.rs

```diff
@@ -95,7 +97,14 @@ async fn validate_proposal_liveness<TYPES: NodeType, I: NodeImplementation<TYPES
     )
     .await;
 
-    if let Err(e) = update_view::<TYPES, I, V>(view_number, event_sender, task_state).await {
+    let epoch_number = TYPES::Epoch::new(epoch_from_block_number(
+        proposal.data.block_header.block_number(),
+        task_state.epoch_height,
+    ));
+
+    if let Err(e) =
+        update_view::<TYPES, I, V>(view_number, epoch_number, event_sender, task_state).await
+    {
```

Comment:
When you merge this with main this will just become a `ViewChange` with the new epoch number, and the changes in `update_view` will just have to be in `handle_view_change`